app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -128,12 +128,57 @@ def process_collection_item(item, key)
       new_result
     end
 
+    def find_host_by_provider_id(provider_id)
+      providers = {
+        :gce => [ManageIQ::Providers::Google::CloudManager, :name],
+        :aws => [ManageIQ::Providers::Amazon::CloudManager, :uid_ems],
+      }
+      instance_uri = URI(provider_id)
+      prov, name_field = providers[instance_uri.scheme.to_sym]
+      return nil unless prov
+      project_name = instance_uri.host
+      _, az, instance_name = instance_uri.path.split('/')
+
+      instance_details = {name_field => instance_name}
+      instance_details[:availability_zone_id] =
+        AvailabilityZone.find_by(:name => az).try(:id) if az != ""
+
+      if project_name
+        project = prov.find_by_project(project_name)
+        return nil unless project
+        prov = project.vms
```

Comment:
@zeari I agree we should simplify this part.

@enoodle can we avoid considering the availability zone and the project?